magenta/music/musicxml_parser.py

```diff
@@ -637,7 +637,7 @@ def _fix_time_signature(self):
       # Insert a new time signature only if it does not equal the global
       # time signature.
       if (self.time_signature is None
-          and fractional_time_signature != fractional_state_time_signature
+          and (fractional_time_signature != fractional_state_time_signature)
```

Comment:
The order of boolean evaluation still isn't as clear as i would like it to be. `and` is has higher binding precedence than `or`, so I think something like this would be more clear:

```
if (pickup_measure or (
    self.time_signature is None and (
      fractional_time_signature != fractional_state_time_signature)):
  ...
```